### PR TITLE
feat: admin module

### PR DESCRIPTION
**Pull request by timlind**
This PR introduces an admin module (reducing tech debt that accumulated in token registry module), with the ability to change admin accounts on the fly. If an account has admin permission it can manage other accounts.

### CODE REVIEW

File: x/admin/types/genesis.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		},
+		{
+			AdminType:    AdminType_ETHBRIDGE,
+			AdminAddress: "sif1syavy2npfyt9tcncdtsdzf7kny9lh777yqc2nd",
+		},
+		{
+			AdminType:    AdminType_TOKENREGISTRY,
+			AdminAddress: "sif1syavy2npfyt9tcncdtsdzf7kny9lh777yqc2nd",
+		},
+	}
```
Comment (canercandan) [inline x/admin/types/genesis.go:62]: @timlind would that mean we need to set one more admin account to the `ADMIN` type?

File: x/tokenregistry/types/genesis.go [hunk 2]
```diff
@@ -58,11 +58,11 @@ func ProdAdminAccounts() *AdminAccounts {
 			},
 			{
 				AdminType:    AdminType_TOKENREGISTRY,
-				AdminAddress: "sif1tpypxpppcf5lea47vcvgy09675nllmcucxydvu",
+				AdminAddress: "sif1npzemsc4s5gxpv2qt3na97tna03cj2h5xxe3cw",
 			},
 			{
 				AdminType:    AdminType_ETHBRIDGE,
-				AdminAddress: "sif1tpypxpppcf5lea47vcvgy09675nllmcucxydvu",
+				AdminAddress: "sif10wgwh7g3jktemd4d8jnswnf0zyk3hsq3uk3tff",
```
Comment (canercandan) [inline x/tokenregistry/types/genesis.go:65]: @timlind I would assume those admin addresses are changed to renew them from the previous one used in too many environments correct?

**Review by canercandan — COMMENTED** [review]
@timlind added some comments

File: x/tokenregistry/types/genesis.go (hunk 2 above)
Comment (timlind) [reply x/tokenregistry/types/genesis.go:65]: Yes, new accounts have been generated here.

File: x/admin/types/genesis.go (hunk 1 above)
Comment (timlind) [reply x/admin/types/genesis.go:80]: Yes, thanks.

File: x/admin/keeper/handler.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func NewHandler(k Keeper) sdk.Handler {
+	msgServer := NewMsgServerImpl(k)
+	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+		switch msg := msg.(type) {
+		case *types.MsgAddAccount:
```
[the remark this answers is not in the document]
Comment (timlind) [reply x/admin/keeper/handler.go:16]: Done

File: x/tokenregistry/keeper/migrations.go [hunk 4]
```diff
@@ -24,19 +24,3 @@ func (m Migrator) MigrateToVer2(ctx sdk.Context) error {
 	m.keeper.SetRegistry(ctx, registry)
 	return nil
 }
-
```
Comment (AustinoBombino) [inline x/tokenregistry/keeper/migrations.go:27]: Will we need a migration here to remove old admin state?

Comment (timlind) [reply x/tokenregistry/keeper/migrations.go:27]: Done.

**Review by AustinoBombino — APPROVED** [review]
nice!

**Comment by canercandan** [comment]
moving this one to 0.13.4 RC 1

**Pull request merged**